baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ForbidJavaxParameterType.java

```diff
@@ -0,0 +1,122 @@
+                + "method which requires newer jakarta types is a runtime error. This check ensures\n"
+                + "that you only supply proper types to these methods which generally just take an\n"
+                + "untyped Object. There is no auto-fix for this check, you must fix it manually")
+public final class ForbidJavaxParameterType extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final Matcher<ClassTree> HAS_JAXRS_ANNOTATIONS =
+            Matchers.hasMethod(MoreMatchers.hasAnnotationWithPackagePrefix("javax.ws.rs"));
+
+    private static final Matcher<ClassTree> IMPLEMENTS_FEATURE = Matchers.isSubtypeOf("javax.ws.rs.core.Feature");
+
+    private static final Matcher<ClassTree> HAS_PATH_ANNOTATION = Matchers.hasAnnotation("javax.ws.rs.Path");
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        List<? extends ExpressionTree> arguments = tree.getArguments();
+        MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
+        List<VarSymbol> typeParameters = methodSymbol.getParameters();
+
+        if (typeParameters == null
+                || typeParameters.isEmpty()
+                || arguments == null
+                || arguments.isEmpty()
+                || arguments.size() != typeParameters.size()) {
+            return Description.NO_MATCH;
+        }
+
+        for (int i = 0; i < typeParameters.size(); ++i) {
+            VarSymbol parameter = typeParameters.get(i);
+            ExpressionTree argument = arguments.get(i);
+
+            Annotation maybeRequireJakarta = parameter.getAnnotation(ForbidJavax.class);
+            if (maybeRequireJakarta == null) {
+                continue;
+            }
+
+            Type resultType = ASTHelpers.getResultType(argument);
+            if (hasJavaxInclusions(resultType, state)) {
+                // we know that resultType is not-null here since hasJavaxInclusions returns false for null
+                return buildDescription(tree)
+                        .setMessage(resultType.asElement().getQualifiedName().toString()
+                                + " registers legacy javax imports but is being supplied to a method which"
+                                + " requires jakarta")
+                        .build();
+            }
+        }
+
+        return Description.NO_MATCH;
+    }
+
+    private boolean hasJavaxInclusions(Type resultType, VisitorState state) {
+        if (resultType == null) {
+            return false;
+        }
+
+        return hasJavaxInclusionsOnType(resultType.asElement(), state);
+    }
+
+    private boolean hasJavaxInclusionsOnType(TypeSymbol symbol, VisitorState state) {
+        if (symbol instanceof ClassSymbol) {
+            ClassSymbol classType = (ClassSymbol) symbol;
+            ClassTree classTree = ASTHelpers.findClass(classType, state);
```

Comment:
I suspect we won't be able to find ClassTrees for most types that are registered as they're defined elsewhere and haven't been parsed by the compiler. Can we use the `Type` object directly with errorprone TypePredicates?

Reply:
Hrm...TypePredicates only give me DescendentOf, so I can get `Feature` but likely could not get anything else.